John/fractional bugs

jordajm opened the pull request:
### Description:

First round of Q4 fractional usage bug fixes

#### Bugs Fixed
- Fixes buyer's date dropdown menus
- Fixes bug on buyer's calendar that blocked off more time than the offers' slots had reserved
- Hides "pending" and "sold" messaging and badges for fractional listings (b/c they can accept multiple offers, so that messaging doesn't make sense)
- Other minor improvements

- @wanderingstan - this also stubs in fractional schemas for "For Rent Housing" and "For Rent Jewelry" as the "2.0.0" versions of those schamas, and the dapp uses those new schemas only if the `ENABLE_FRACTIONAL` env var is set to true. 

So, we should be able save ourselves some hassle by merging these schemas in and use them for local dev without messing up the production dapp.

### Checklist:

- [x] Test your work and double-check to confirm that you didn't break anything
- [ ] Wrap any new text/strings for translation
- [ ] Map any new environment variables with a default value in the Webpack config
- [ ] Update any relevant READMEs and [docs](https://github.com/OriginProtocol/origin/tree/master/origin-docs)

jordajm commented:
@wanderingstan - it's not thoroughly tested yet, but when you get a chance, pull the latest on this branch and check out the new UI for the seller's calendar... click "create listing", then "for rent" then "housing" to see it. LMK if you have ideas/feedback/bug reports/etc

wanderingstan commented:
Great @jordajm ! Not sure if hourly is even supposed to be working, but played with it too. 

Quick notes:

### DAILY
- When setting availability, will let you select ranges in the past, and enter price, but won't display them. 
- Think we need a UI way to get back to "today". (And should we show year? I was able to get myself lost in dates!)
- Can't edit a listing: it show's UI with per-listing Price, and gives "Please fix errors before continuing" no matter what you do. 

### HOURLY
- When creating a hourly (jewelery) listing, when marking availability, created ranges don't appear. 
  ...BUT, they do appear on buyers calendar after listing is created!
- Transaction status page is old style

jordajm commented:
@wanderingstan other than some hairy merge conflicts which I don't have the steam to sort through tonight (but will do tomorrow evening), this PR is ready for review/merge. Aside from the impending dooom... er, I mean slot protocol changes, fractional usage would be ready to deploy to staging and test 🎉 🤦‍♂️ 😬

wanderingstan commented:
Ah it won't be that bad...no doom!

jordajm commented:
@wanderingstan - merge conflicts fixed - ready for final review/merge. 

Note - this PR still won't change anything in the dapp unless the ENABLE_FRACTIONAL env var is set.

I'll start a new PR for the slot protocol changes... I agree - it shouldn't be that bad ;-)

wanderingstan commented:
When I attempt to make an offer on some time, I get a validation error around a missing `timeZone` property. I assume this is part of transitioning to jCal? 

```
app.024b8901.js:579649 Error: Data failed schema validation.
        Schema id: https://schema.originprotocol.com/offer_1.0.0.json
        Data: {"listingId":"999-000-5","listingType":"fractional","totalPrice":{"amount":"1.00000","currency":"ETH"},"commission":{"amount":"0","currency":"OGN"},"finalizes":31536000,"slots":[{"startDate":"2019-01-07T07:00:00.000Z","endDate":"2019-01-08T06:59:59.000Z","price":{"currency":"ETH","amount":"0.5"}},{"startDate":"2019-01-08T07:00:00.000Z","endDate":"2019-01-09T06:59:59.000Z","price":{"currency":"ETH","amount":"0.5"}}],"schemaId":"https://schema.originprotocol.com/offer_1.0.0.json"}.
        Errors: [{"keyword":"required","dataPath":"","schemaPath":"#/definitions/unitOffer/required","params":{"missingProperty":"unitsPurchased"},"message":"should have required property 'unitsPurchased'"},{"keyword":"required","dataPath":".slots[0]","schemaPath":"#/required","params":{"missingProperty":"timeZone"},"message":"should have required property 'timeZone'"},{"keyword":"required","dataPath":".slots[1]","schemaPath":"#/required","params":{"missingProperty":"timeZone"},"message":"should have required property 'timeZone'"},{"keyword":"oneOf","dataPath":"","schemaPath":"#/oneOf","params":{"passingSchemas":null},"message":"should match exactly one schema in oneOf"}]
```

jordajm commented:
@wanderingstan I fixed the validation bug with the offers. Good catch, thanks 👍 

For clicking to create an event, I'm not sure how else we'd handle that - do you have suggestions? 
If the user does mistakenly create an event by clicking the calendar on the wrong date or something, they just have to click the "delete" button on the upper right of the black box beside the calendar. 

I also fixed a few random issues I found on my-purchase, offer detail, purchase detail, etc. Check those pages out too when you get a chance.

wanderingstan commented:
For the click-create issue, I'd say that nothing should get created until the user picks a price and hits "Save". (We should also prevent a listing price of zero ETH in any case.)

Or put another way: clicking or dragging on the calendar should only be a _selection_ process. Creation only happens when they explicitly hit save.
